docs/src/main/asciidoc/kafka-schema-registry-avro.adoc

```diff
@@ -0,0 +1,604 @@
+                .execute(
+                        () -> {
+                            while (true) {
+                                given()
+                                        .contentType(ContentType.JSON)
+                                        .body("{\"title\":\"The Shawshank Redemption\",\"year\":1994}")
+                                    .when().post("/movies")
+                                    .then()
+                                        .statusCode(202);
+                                given()
+                                        .contentType(ContentType.JSON)
+                                        .body("{\"title\":\"12 Angry Men\",\"year\":1957}")
+                                    .when().post("/movies")
+                                    .then()
+                                        .statusCode(202);
+                                try {
+                                    Thread.sleep(200L);
+                                } catch (InterruptedException e) {
+                                    break;
+                                }
+                            }
+                        }
+                );
+        return executorService;
+    }
+
+}
+----
+
+== Avro code generation details
+In this guide we used the Quarkus code generation mechanism to generate Java files
+from Avro schema.
+
+Under the hood, the mechanism uses `org.apache.avro:avro-compiler`.
+
+You can use the following configuration properties to alter how it works:
+
+- `avro.codegen.[avsc|avdl|avpr].imports` - a list of files or directories that should be compiled first thus making them
+importable by subsequently compiled schemas. Note that imported files should not reference each other. All paths should be relative
+to the `src/[main|test]/avro` directory. Passed as a comma-separated list.
+- `avro.codegen.stringType` - the Java type to use for Avro strings. May be one of `CharSequence`, `String` or
+`Utf8`. Defaults to `String`
+- `avro.codegen.createOptionalGetters` - enables generating the `getOptional...`
+methods that return an Optional of the requested type. Defaults to `false`
+- `avro.codegen.enableDecimalLogicalType` - determines whether to use Java classes for decimal types, defaults to `false`
+- `avro.codegen.createSetters` - determines whether to create setters for the fields of the record.
+Defaults to `false`
+- `avro.codegen.gettersReturnOptional` - enables generating `get...` methods that
+return an Optional of the requested type. Defaults to `false`
+- `avro.codegen.optionalGettersForNullableFieldsOnly`, works in conjunction with `gettersReturnOptional` option.
+If it is set, `Optional` getters will be generated only for fields that are nullable. If the field is mandatory,
+regular getter will be generated. Defaults to `false`
+
+== Further reading
+
+* link:https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/2/kafka/kafka.html[SmallRye Reactive Messaging Kafka] documentation
```

Comment:
```suggestion
* link:https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/2.9/kafka/kafka.html[SmallRye Reactive Messaging Kafka] documentation
```

That's a problem I have with Antora. It references a very old version (2 -> 2.0) while we are 2.9 and Avro support has been improved in 2.5.